src/Version/CMakeLists.txt

```diff
@@ -53,4 +53,31 @@ set_source_files_properties("${version_inc}"
              HEADER_FILE_ONLY TRUE
   )
 
-add_custom_target(OMVersion ALL DEPENDS ${version_inc})
+add_onnx_mlir_library(OMVersion
+  Version.cpp
+  ${version_inc}
+
+  LINK_LIBS PUBLIC
+    LLVMSupport
+)
+
+file(READ "${CMAKE_SOURCE_DIR}/VERSION_NUMBER" ONNX_MLIR_VERSION)
+string(STRIP "${ONNX_MLIR_VERSION}" ONNX_MLIR_VERSION)
+list(APPEND DEFINITIONS "ONNX_MLIR_VERSION=\"${ONNX_MLIR_VERSION}\"")
+execute_process(
+  COMMAND cat VERSION_NUMBER
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/onnx
+  OUTPUT_VARIABLE ONNX_VERSION
```

Comment:
@whitneywhtsang I think it's OK to use the VERSION_NUMBER file although it would be helpful to figure out why the git command failed on the Mac build. I looked into this a bit and it appears to me that the github action `checkout@v2` is not cloning the tags properly, which appears to be a known problem https://github.com/actions/checkout/issues/206. I would suggest try the `checkout@v3` action and failing that try the proposed fix:
```
diff --git a/.github/workflows/macos-amd64-build.yml b/.github/workflows/macos-amd64-build.yml
index 8e69a516..5a8b9b8b 100644
--- a/.github/workflows/macos-amd64-build.yml
+++ b/.github/workflows/macos-amd64-build.yml
@@ -9,6 +9,8 @@ jobs:
     - uses: actions/checkout@v2
       with:
         submodules: recursive
+    - run: |
+        git fetch --prune --unshallow --tags
     - uses: actions/setup-python@v2
       with:
         python-version: '3.9'
```
And if this fixes the problem, we should update our `macos-amd64-build.yml`.

Reply:
@whitneywhtsang I see that the command `git fetch --prune --unshallow --tags` failed. Try without the `--prune` option.

Reply:
For the record, `checkout@v3` doesn't work:
```
--- a/.github/workflows/macos-amd64-build.yml
+++ b/.github/workflows/macos-amd64-build.yml
@@ -6,7 +6,7 @@ jobs:
   build:
     runs-on: macos-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         submodules: recursive
```

Removing `--prune` still fails.

Reply:
Sigh. OK one last try. Let's just do fetch in third_party/onnx. If this still doesn't work, we will just use VERSION_NUMBER.
```
diff --git a/.github/workflows/macos-amd64-build.yml b/.github/workflows/macos-amd64-build.yml
index 8e69a516..0ac3c8bb 100644
--- a/.github/workflows/macos-amd64-build.yml
+++ b/.github/workflows/macos-amd64-build.yml
@@ -30,6 +30,8 @@ jobs:
         sh ~/work/onnx-mlir/onnx-mlir/utils/build-mlir.sh
     - name: install third_party/onnx
       run: |
+        cd ~/work/onnx-mlir/onnx-mlir/third_party/onnx
+        git fetch --prune --unshallow --tags
         cd ~/work/onnx-mlir/onnx-mlir
         pip3 install -e third_party/onnx
     - name: build onnx-mlir
```

Reply:
`git fetch --prune --unshallow --tags`
or `git fetch --unshallow --tags`
or `git fetch --prune --tags`
or `git fetch --tags`
are failing with:
```
fatal: remote error: upload-pack: not our ref b57c11391d9c749d6ef2b6139a339acac69bbb76
Errors during submodule fetch:
	third_party/benchmark
Error: Process completed with exit code 1.
```

Reply:
Fetch in third_party/onnx works.

However, I prefer the following change, as it will make getting onnx-mlir and onnx version consistent. 
```
diff --git a/src/Version/CMakeLists.txt b/src/Version/CMakeLists.txt
index 316d0e32..50c28ca6 100644
--- a/src/Version/CMakeLists.txt
+++ b/src/Version/CMakeLists.txt
@@ -64,17 +64,11 @@ add_onnx_mlir_library(OMVersion
 file(READ "${CMAKE_SOURCE_DIR}/VERSION_NUMBER" ONNX_MLIR_VERSION)
 string(STRIP "${ONNX_MLIR_VERSION}" ONNX_MLIR_VERSION)
 list(APPEND DEFINITIONS "ONNX_MLIR_VERSION=\"${ONNX_MLIR_VERSION}\"")
-execute_process(
-  COMMAND git tag --points-at HEAD
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/onnx
-  OUTPUT_VARIABLE ONNX_VERSION
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
+file(READ "${CMAKE_SOURCE_DIR}/third_party/onnx/VERSION_NUMBER" ONNX_VERSION)
+string(STRIP "${ONNX_VERSION}" ONNX_VERSION)
 if ("${ONNX_VERSION}" STREQUAL "")
   message(WARNING "third_party/onnx version not detected, you may be on the wrong commit hash")
 else()
-  # Strip leading "v"
-  STRING(REGEX REPLACE "^v" "" ONNX_VERSION ${ONNX_VERSION})
   message(STATUS "Onnx version             : ${ONNX_VERSION}")
   list(APPEND DEFINITIONS "ONNX_VERSION=\"${ONNX_VERSION}\"")
```

@gongsu832 What do you think?

Reply:
@whitneywhtsang This is fine since onnx provides such a version file. Just keep in mind that most code probably won't. Also, let's keep the change to `macos-amd64-build.yml` so tags are fetched. It's a bit surprising that the checkout action doesn't do this by default. Maybe there is some option we have to specify but we can worry about that later. Thanks for trying all the combinations.

Reply:
Thanks @gongsu832